app/Listeners/TaskUpdateMessage.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace App\Listeners;
+
+use App\Events\TaskUpdate;
+use App\GenericModel;
+use App\Profile;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Contracts\Queue\ShouldQueue;
+
+class TaskUpdateMessage
+{
+
+    protected $tasks;
+
+    /**
+     * Create the event listener.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        //
+    }
+
+    /**
+     * Handle the event.
+     *
+     * @param  TaskUpdate $event
+     * @return void
+     */
+    public function handle(TaskUpdate $event)
+    {
+        $this->tasks = $event->tasks;
+
+        //task user id
+        $user_id = $this->tasks->task_history[0]['user'];
+
+        $task_xp = $this->tasks->xp;
+
+        $profile = Profile::find($user_id);
+
+        //avoid milliseconds
+        $submitted = $this->tasks->due_date - intval($this->tasks->task_history[0]['timestamp'] / 1000);
+
+        $coefficient = ($this->tasks->due_date - $submitted) / $this->tasks->due_date;
+        if ($profile->id == $user_id) {
+            switch ($coefficient) {
+                case ($coefficient <= 0.75):
```

Comment:
This is also true for negative, where if someone misses deadline you'd award them with 3 xp